src/client.ts

```diff
@@ -8963,6 +8965,11 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             },
         );
     }
+
+    public dumpDebugLogs() {
+        logger.debug('matrix-js-sdk: dumping extra debug logs to the console');
+        this.emit(ClientEvent.DumpDebugLogs);
+    }
```

Comment:
using logs is a really bad mechanism given that it relies on consumers monkey-patching `console` which we don't say anywhere in the generated docs to do

Reply:
This doesn't have to live in `matrix-js-sdk`. What event emitter pattern can we use exclusively in `matrix-react-sdk`? This point is tracked by https://github.com/matrix-org/matrix-react-sdk/pull/8502#discussion_r865555576